crates/sui-graphql-rpc/src/types/query.rs

```diff
@@ -46,9 +54,112 @@ impl Query {
     }
 
     // availableRange - pending impl. on IndexerV2
-    // dryRunTransactionBlock
     // coinMetadata
 
+    /// Simulate running a transaction to inspect its effects without
+    /// committing to them on-chain.
+    ///
+    /// `txBytes` either a `TransactionData` struct or a `TransactionKind`
+    ///     struct, BCS-encoded and then Base64-encoded.  The expected
+    ///     type is controlled by the presence or absence of `txMeta`: If
+    ///     present, `txBytes` is assumed to be a `TransactionKind`, if
+    ///     absent, then `TransactionData`.
+    ///
+    /// `txMeta` the data that is missing from a `TransactionKind` to make
+    ///     a `TransactionData` (sender address and gas information).  All
+    ///     its fields are nullable: `sender` defaults to `0x0`, if
+    ///     `gasObjects` is not present, or is an empty list, it is
+    ///     substituted with a mock Coin object, and `gasPrice` defaults to
+    ///     the reference gas price.
+    ///
+    /// `skipChecks` optional flag to disable the usual verification
+    ///     checks that prevent access to objects that are owned by
+    ///     addresses other than the sender, and calling non-public,
+    ///     non-entry functions.  Defaults to false.
+    ///
+    /// `epoch` the epoch to simulate executing the transaction in.
+    ///     Defaults to the current epoch.
+    async fn dry_run_transaction_block(
+        &self,
+        ctx: &Context<'_>,
+        tx_bytes: String,
+        tx_meta: Option<TransactionMeta>,
+        skip_checks: Option<bool>,
+        epoch: Option<u64>,
+    ) -> Result<DryRunResult> {
+        let skip_checks = skip_checks.unwrap_or(false);
+
+        let sui_sdk_client: &Option<SuiClient> = ctx
```

Comment:
Apologies for potentially a noob question, but how will this work when RPC 1.0 will be deprecated and the SDK will move over to GraphQL service? The used `read_api` here will change, so I am not sure this is the right way to implement this.

Reply:
Before we fully deprecate JSON-RPC we will need to separate the quorum driver into its own service (or integrate it into the GraphQL service, at which point we can avoid taking the dependency on the SDK.